weatherman.py

```diff
@@ -1,214 +1,362 @@
+                min_temp_dict[year] = month_min_temp
+        else:
+            # If the result get from function is None
+            pass
+
+        # If result from function is not None
+        if res_from_min_humidity != 10000:
+            # e.g result = [1996-12-1,32]
+            date = res_from_min_humidity[0]
+            # Split the date so that we can use Year e.g 1996 as a key
+            year = str(date)[:4]
+            # Store the MINHumidity per month
+            month_min_humidity = res_from_min_humidity[1]
+            if year in min_humidity_dict.keys():
+                # If key(Year) is already store in dictionary
+                i = int(month_min_humidity)
+                j = int(min_humidity_dict[year])
+                # Compare the NewHumidity with existed Humidity
+                if i < j:
+                    # If NewHumidity is less than existed Humidity.
+                    # Replace it with NewHumidity
+                    min_humidity_dict[year] = month_min_humidity
+                else:
+                    # If NewHumidity is greater than existed Humidity.
+                    pass
+            else:
+                # If key is not already exist.Add it in dictionary with Year as a key
+                min_humidity_dict[year] = month_min_humidity
+        else:
+            # If result from function is None
+            pass
+
+        # If result from function is not 0
+        if res_from_max_humidity != 0:
+            # e.g result = [1996-12-1,32]
+            date = res_from_max_humidity[0]
+            # Split the date so that we can use Year e.g 1996 as a key
+            year = str(date)[:4]
+            # Store the MAXHumidity per month
+            month_max_humidity = res_from_max_humidity[1]
+            if year in max_humidity_dict.keys():
+                # If key(Year) is already store in dictionary
+                i = int(month_max_humidity)
+                j = int(max_humidity_dict[year])
+                # Compare the NewHumidity with existed Humidity
+                if i < j:
+                    # If NewHumidity is greater than existed Humidity.
+                    # Replace it with NewHumidity
+                    max_humidity_dict[year] = month_max_humidity
+                else:
+                    # If NewHumidity is less than existed Humidity.
+                    pass
+            else:
+                # If key is not already exist.Add it in dictionary with Year as a key
+                max_humidity_dict[year] = month_max_humidity
+        else:
+            # If result from function is None
+            pass
+
+# MAIN PROGRAM
+
+report_type = str(input("For Annual Report :1\n"
+                        "For Hottest Days Report:2\n"
+                        "Enter Report Type: "
+                        )
+                  )
+# Take input from user. What type of report to display
+path = "*.txt"

```

Comment:
files are in current Project Directory. We just have to extract _*.txt_ files from folder.